test/extended/openstack/servergroup.go

```diff
@@ -121,8 +134,60 @@ var _ = g.Describe("[sig-installer][Feature:openstack] The OpenStack platform",
 		}
 		o.Expect(host_ids).To(o.HaveLen(len(masterInstanceUUIDs)),
 			"Master nodes should be on different hosts when anti-affinity policy is used")
+	})
+
+	g.It("creates Worker nodes in a server group", func() {
+		g.By("Getting the Worker instances' Server groups")
+		{
+			for _, item := range workerNodeList.Items {
+				machineAnnotation := strings.SplitN(item.Annotations["machine.openshift.io/machine"], "/", 2)
+				o.Expect(machineAnnotation).To(o.HaveLen(2))
+
+				res, err := ms.Namespace(machineAnnotation[0]).Get(ctx, machineAnnotation[1], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				instanceWorkerGroupNameField := getFromUnstructured(res, "spec", "providerSpec", "value", "serverGroupName")
+				o.Expect(instanceWorkerGroupNameField).NotTo(o.BeNil(), "the Server group name should be present in the Machine definition")
+
+				instanceWorkerGroupName := instanceWorkerGroupNameField.(string)
+				o.Expect(instanceWorkerGroupName).NotTo(o.BeEmpty(), "the Server group name should not be the empty string")
+				if workerGroupName == "" {
+					workerGroupName = instanceWorkerGroupName
+				} else {
+					o.Expect(instanceWorkerGroupName).To(o.Equal(workerGroupName), "two Control plane Machines have different controlPlaneGroupName set")
+				}
+			}
+		}
+		g.By("Checking the actual members of the Server group")
+		{
+			serverGroupsWithThatName, err := serverGroupIDsFromName(computeClient, workerGroupName)
+			o.Expect(serverGroupsWithThatName, err).To(o.HaveLen(1), "the server group name either was not found or is not unique")
+
+			serverGroup, err := servergroups.Get(computeClient, serverGroupsWithThatName[0]).Extract()
+			o.Expect(serverGroup.Members, err).To(o.ContainElements(workerInstanceUUIDs...))
+		}
 
 	})
+
+	// OCP 4.10: https://issues.redhat.com/browse/OSASINFRA-2507
```

Comment:
I think it's `OCP 4.10: https://issues.redhat.com/browse/OSASINFRA-2570`

Reply:
Done